apps/files/js/file-upload.js

```diff
@@ -997,7 +1001,8 @@ OC.Uploader.prototype = _.extend({
 						freeSpace = $('#free_space').val()
 					} else if (upload.getTargetFolder().indexOf(self.fileList.getCurrentDirectory()) === 0) {
 						// Check subdirectory free space if file is uploaded there
-						var targetSubdir = upload._targetFolder.replace(self.fileList.getCurrentDirectory(), '')
+						// Retrieve the folder destination name
+						var targetSubdir = upload._targetFolder.split('/').pop()
```

Comment:
this seems to only take the last component of the folder.

what if targetFolder has more components like "a/b/c/d" ?

Reply:
Correct. That seemed to have been the issue prior to it. 

In the following line, `upload.uploader.fileList` represents the fileList in the current folder. As such, via drag and drop you can only be dragging to either <current folder> or a subfolder of the current folder. 

Assuming you are in folder /a/b/c
and you're dragging a file straight into 'd'

`upload.uploader.fileList` will have all the files in /a/b/c .

Calling `upload.uploader.fileList.getModelForFile('a/b/c/d')` will return null, so 
`upload.uploader.fileList.getModelForFile(targetSubdir).get('quotaAvailableBytes')` will raise an exception.
Calling `upload.uploader.fileList.getModelForFile('d')` will return the model for the folder a/b/c/d, and 
`upload.uploader.fileList.getModelForFile(targetSubdir).get('quotaAvailableBytes')` will return the quota for the subfolder we're dragging the file into.

Was that the case you were bringing up? If not, let me know - as I stated, you know more about this code than I do, I am just trying to help here.

Thank you for your time.

Reply:
Correct. That seemed to have been the issue prior to it. 

In the following line, `upload.uploader.fileList` represents the fileList in the current folder. As such, via drag and drop you can only be dragging to either <current folder> or a subfolder of the current folder. 

Assuming you are in folder /a/b/c
and you're dragging a file straight into 'd'

`upload.uploader.fileList` will have all the files in /a/b/c .

Calling `upload.uploader.fileList.getModelForFile('a/b/c/d')` will return null, so 
`upload.uploader.fileList.getModelForFile(targetSubdir).get('quotaAvailableBytes')` will raise an exception.
Calling `upload.uploader.fileList.getModelForFile('d')` will return the model for the folder a/b/c/d, and 
`upload.uploader.fileList.getModelForFile(targetSubdir).get('quotaAvailableBytes')` will return the quota for the subfolder we're dragging the file into.

Was that the case you were bringing up? If not, let me know - as I stated, you know more about this code than I do, I am just trying to help here.

Thank you for your time.

Reply:
Yes, that's the point, we use the subfolder name when it's dropped into a subfolder

Reply:
Thanks. This PR makes it indeed use the subfolder rather than the full path, which was causing problems when checking for available quota.

Reply:
Or would you be more comfortable with 

```
upload._targetFolder.replace(self.fileList.getCurrentDirectory()+'/', '')
```

?

That also works. I just don't have a scenario where the `targetSubdir` is not an immediate child of the current folder, hence using pop(), but let me know if there's a preference in the code.

Reply:
Yes please, it feels less error-prone :)

```suggestion
						var targetSubdir = upload._targetFolder.replace(self.fileList.getCurrentDirectory()+'/', '')
```

Reply:
Thanks! Actually, I just tested this but this actually breaks when we're at the root folder.

When in a subfolder, 
`self.fileList.getCurrentDirectory()` is, say, "/parent", and 
targetFolder is "/parent/target", so 
we want targetSubdir to be "target" only. 
`self.fileList.getCurrentDirectory()+'/'` is "/parent/", which matches and replaces the right string.
This works.

But if in the root folder, 
`self.fileList.getCurrentDirectory()` is, "/", and 
targetFolder is "/target", so 
we want targetSubdir to be "target" only. 
`self.fileList.getCurrentDirectory()+'/'` becomes "//", which doesn't find a match, and so everything breaks from then on.

I am happy to come up with an `if` condition there, if you feel strongly that we should have the latter version, but thought I'd raise this before sending over another change. :)